app/code/core/Mage/Catalog/Model/Product/Attribute/Api.php

```diff
@@ -217,10 +217,10 @@ public function update($attribute, $data)
             $model->save();
             // clear translation cache because attribute labels are stored in translation
             Mage::app()->cleanCache(array(Mage_Core_Model_Translate::CACHE_TAG));
-            return true;
         } catch (Exception $e) {
             $this->_fault('unable_to_save', $e->getMessage());
         }
+        return true;
```

Comment:
thats a very strong behavior change to now return true even in case of an error?

Reply:
Is it? `$this->_fault('unable_to_save', $e->getMessage());` throws an exception internally, so return is not reached.

Reply:
see https://github.com/OpenMage/magento-lts/blob/d19f0e13d3d2fcbcc2d8fa6073568d5f95442470/app/code/core/Mage/Api/Model/Resource/Abstract.php#L97

Reply:
@Flyingmana agree to _return is not reached_.